File: buerguer-pig/AMarotoChivite-BurguerPig/AMarotoChivite-BurguerPig-Java/src/main/java/Ficheros/BurguerPig/mappers/IngredienteMapper.java
package Ficheros.BurguerPig.mappers;

import Ficheros.BurguerPig.mappers.base.IDtoMapper;
import Ficheros.BurguerPig.models.Ingredient;
import Ficheros.BurguerPig.models.dto.IngredienteDTO;

public class IngredienteMapper implements IDtoMapper<IngredienteDTO, Ingredient> {

    @Override
    public IngredienteDTO toDto(Ingredient model) {
        return new IngredienteDTO(
                model.getName(),
                Double.toString(model.getPrice()),
                Integer.toString(model.getID())
        );
    }

    @Override
    public Ingredient toModel(IngredienteDTO dto) {
        return new Ingredient(
                dto.getName(),
                Double.parseDouble(dto.getPrice()),
                Integer.parseInt(dto.getId())
        );
    }
}

File: buerguer-pig/SergioDeIscar-restaurante-kotlin/RestauranteFicheros/src/main/kotlin/mappers/IngredienteMapper.kt
package mappers

import dto.IngredienteDto
import locate.moneyToFloat
import locate.toLocalMoney
import models.Ingrediente

fun Ingrediente.toDto() = IngredienteDto(
    id = this.id.toString(),
    nombre = this.nombre,
    precio = this.precio.toLocalMoney()
)

fun IngredienteDto.toClass() = Ingrediente(
    id = this.id.toLong(),
    nombre = this.nombre,
    precio = this.precio.moneyToFloat()
)